src/lib/storage/storage_manager.cpp

```diff
@@ -29,7 +29,10 @@ void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> t
     Assert(table->get_chunk(chunk_id)->has_mvcc_data(), "Table must have MVCC data.");
   }
 
+  // Create table statistics and chunk pruning statistics for added table.
   table->set_table_statistics(TableStatistics::from_table(*table));
+  generate_chunk_pruning_statistics(table);
```

Comment:
Why do we have one version where we check for existing statistics and one without? Should generate_chunk_pruning_statistics fail if statistics are already present?

Reply:
Good point. Moved the check into the actual `generate_chunk_pruning_statistics(chunk)` method and removed the check from other places.

Does that clear it up?